python_modules/dagster/dagster/_core/definitions/sensor_definition.py

```diff
@@ -197,10 +200,22 @@ def __init__(
         last_run_key: Optional[str],
         cursor: Optional[str],
         repository_name: Optional[str],
+        repository_def: "RepositoryDefinition",
         asset_keys: Sequence[AssetKey],
         instance: Optional[DagsterInstance] = None,
     ):
+        self.repository_def = repository_def
```

Comment:
Class attributes should generally be private (start with an underscore), so users can't overwrite them.